internal/templates/template.go

```diff
@@ -23,16 +25,23 @@ func Init(w io.Writer, template, dir string) error {
 	}
 
 	templatePath, ok := templates[template]
+	cloneURL := "https://github.com/jetpack-io/devbox.git"
 	if !ok {
-		return usererr.New("unknown template %q", template)
+		u, err := url.Parse(template)
+		splits := strings.Split(u.Path, "/")
+		if err != nil || u.Scheme != "https" || len(splits) == 0 {
+			return usererr.New("unknown template name or format %q", template)
+		}
+		templatePath = splits[len(splits)-1]
+		cloneURL = template
```

Comment:
I would to a tiny bit of validation here. Something like:

```golang
if !ok {
  cloneURL, templatePath, err := parseGithubURL(template)
  if err != nil {
    return err
  }
}

func parseGithubURL(u string) (string, string, error) {
  // validate u 
  // parse 
}

Reply:
Putting `parseGithubURL` let's you write a few unit tests as well to ensure you are getting what you expect.